### ISSUE TITLE
Dismissing modal view controller causes the screen to black out - iOS 8

### ISSUE THREAD

**Issue by adityatrivedi** [TeehanLax/UIViewController-Transitions-Example#5]
Reproduction Steps:
1) Launch the app
2) Tap on "Present view controller"
3) Tap the "done" button
4) Black screen appears

I think I found the place where the problem seems to be. 
In iOS 8 [transitionContext completeTranstion:YES] seems to be doing something weird. If you comment that line of code out, you can see the destination view controller, but cannot interact with it.

**Comment by ashfurrow** [comment]
This is due to a serious bug in the current betas of iOS 8. I've filed a [radar](http://openradar.appspot.com/radar?id=5320103646199808), which you can duplicate. In the mean time, there is a [workaround](http://stackoverflow.com/questions/24338700/from-view-controller-disappears-using-uiviewcontrollercontexttransitioning). Let me know if you have any questions.

**Comment by marianoabdala** [comment]
I'm _so_ glad I found this. Workaround works great, thanks!

**Comment by maxhs** [comment]
This has been very frustrating to debug. Great catch/observation(s).

**Comment by BlakeLucchesi** [comment]
Thank you so much for posting the SO link. Saved my sanity!

**Comment by victorBaro** [comment]
[This solution](http://stackoverflow.com/a/25655575/2178888) worked for me

**Comment by gautamjain987** [comment]
This worked for me.
[[UIApplication sharedApplication].keyWindow addSubview:toViewController.view];
toViewController.view.userInteractionEnabled = YES;

**Comment by rainypixels** [comment]
FYI — I just upgraded to XCode 6.1 GM2 / iOS 8.1 B2, and the workaround isn't needed anymore.

**Comment by heitor-canabal-movile** [comment]
@rainypixels I upgraded XCode 6.1 and it did not work for me. Can you show your code, please?

**Comment by carguezu** [comment]
You also can check this PR that solves the issue: https://github.com/TeehanLax/UIViewController-Transitions-Example/pull/6

Or check this example: https://github.com/carguezu/CGZTransitions

**Comment by pbassut** [comment]
@carguezu weird. The only thing I did was setting the modalPresentationStyle before actually presenting the viewController to get the transparent background, like this:

```
    InfoViewController *info = [[InfoViewController alloc] initWithNibName:@"InfoViewController" bundle:nil];
    info.modalPresentationStyle = UIModalPresentationCustom;
    [self presentViewController:info animated:YES completion:nil];
```

It's weird because I've tried that before, and it didn't work.

**Comment by kingja01** [comment]
If you use custom view controller transition style. The IOS8 fixed some bugs in IOS7.
U don't need to add fromViewController.view anymore.
-------------------Don't use this.
//[transitionContext.containerView addSubview:fromViewController.view];
//[transitionContext.containerView addSubview:toViewController.view];
//[transitionContext.containerView addSubview:fromViewController.view];

-------------------Fix it like this
UIView *containerView = [transitionContext containerView];
UIViewController *fromViewController = [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
UIViewController *toViewController = [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];

// Boolean value to determine presentation or dismissal animation
if (self.presenting){  
    [transitionContext.containerView addSubview:toViewController.view];
    // Your presenting animation code
} else {
    // Your dismissal animation code
}

Check the solution from DJSK.
http://stackoverflow.com/questions/25588617/ios-8-screen-blank-after-dismissing-view-controller-with-custom-presentation

**Comment by carguezu** [comment]
@kingja01 dude, that is the same as I mentioned before 👍

**Comment by nihilenz** [comment]
I tried the solution provided by @kingja01 and everything seems ok using the traditional UIKit animations version.
But using the UIKit Dynamics version I have some issues with non-interactive dismiss (Dismiss button).

The button stops working in some situations (if you are quick to open and dismiss) and the app remains stuck in this state.

In this example the button stops working with no warning, in my app with "Warning: Attempt to dismiss from view controller <ViewControllerClass> while a presentation or dismiss is in progress!"

Any idea?

**Comment by ashfurrow** [comment]
Hmm, I've not seen that, but I have not extensively worked with custom transitions in iOS 8, and there are some differences. Could you post a sample project demonstrating the problem so we could take a look?

**Comment by nihilenz** [comment]
https://github.com/nihilenz/UIViewController-Transitions-Example

If you are quick to open the left view controller with interactive transition and to tap on the "Dismiss" button, the app remains stuck and a warning can be shown (UIViewController-Transitions-Example[4585:676902] Warning: Attempt to dismiss from view controller <UINavigationController: 0x12ee0d550> while a presentation or dismiss is in progress!)

Thank you very much!

**Comment by ashfurrow** [comment]
@nihilenz Sorry I haven't had a chance to look at that – been very busy. I'll investigate soon.

**Comment by ashfurrow** [comment]
@nihilenz I was able to reproduce the issue you mentioned. The fix is pretty simple: we just need to disable interactions on the `toViewController`'s view as well. That way, the dismiss button cannot be interacted with until the view animation comes to a complete stop. https://github.com/ashfurrow/UIViewController-Transitions-Example/compare/nihilenz:master...ashfurrow:master

**Comment by nihilenz** [comment]
@ashfurrow Thank you very much! 👍

**Comment by hiyali** [comment]
remove parent view `translatesAutoresizingMaskIntoConstraints = false` this line

**Comment by horseshoe7** [comment]
This is what solved it for me.  You need to change all the fromViewController.view to accessing the view via viewForKey: 

http://stackoverflow.com/a/25193675/421797

**Comment by romeugodoi** [comment]
Has anyone fixed it? Here I have the same issue when I switch between 2 tabs and one of them has another VC presented over current context.

**Comment by ashfurrow** [comment]
No, and I don't suspect anyone ever will. [Teehan+Lax](https://teehanlax.com) is no longer operating and no one has push rights to its repos anymore. The issues are here for you to search through and figure things out, though. Good luck.

**Comment by LanceSamaria** [comment]
@romeugodoi did you find a fix? I'm having the tab switch problem now

**Comment by patelnirav48** [comment]
> @carguezu weird. The only thing I did was setting the modalPresentationStyle before actually presenting the viewController to get the transparent background, like this:
> 
> ```
>     InfoViewController *info = [[InfoViewController alloc] initWithNibName:@"InfoViewController" bundle:nil];
>     info.modalPresentationStyle = UIModalPresentationCustom;
>     [self presentViewController:info animated:YES completion:nil];
> ```
> 
> It's weird because I've tried that before, and it didn't work.

weird but this works for me... Thanks a lot
